Gems/Atom/RPI/Code/Source/RPI.Public/Scene.cpp

```diff
@@ -695,6 +695,13 @@ namespace AZ
                 }
             }
 
+            // the pipeline states might have changed during the OnStartFrame, rebuild the lookup
+            if (m_pipelineStatesLookupNeedsRebuild)
+            {
+                RebuildPipelineStatesLookup();
+                m_pipelineStatesLookupNeedsRebuild = false;
```

Comment:
Just a small suggestion, I would move
`m_pipelineStatesLookupNeedsRebuild = false;`
into the RebuildPipelineStatesLookup() function itself in case someone adds another call to RebuildPipelineStatesLookup in the future, that way we'd avoid potential double calls to the function.